api/all/src/main/java/io/opentelemetry/api/GlobalOpenTelemetry.java

```diff
@@ -219,15 +223,29 @@ private static OpenTelemetry maybeAutoConfigureAndSetGlobal() {
       return null;
     }
 
+    // If autoconfigure module is present but global autoconfigure disabled log a warning and return
+    boolean globalAutoconfigureEnabled =
+        Boolean.parseBoolean(ConfigUtil.getString(GLOBAL_AUTOCONFIGURE_ENABLED_PROPERTY));
+    if (!globalAutoconfigureEnabled) {
+      logger.log(
+          Level.INFO,
+          "AutoConfiguredOpenTelemetrySdk found on classpath but automatic configuration is disabled."
+              + " To enable, run your JVM with -D"
+              + GLOBAL_AUTOCONFIGURE_ENABLED_PROPERTY
+              + "=true");
+      return null;
```

Comment:
is null the right response here, rather than the no-op implementation?

Reply:
Looking at the caller of this...I think we might be doing the wrong thing right now. If this method returns null, we're calling `set` with the no-op impl. That will mean (I think?) that we will always end up returning the no-op, because of the check in the `set` method. I think this is wrong because we do want people to be able to `set` after calling a uninitialized `get`.

Reply:
> That will mean (I think?) that we will always end up returning the no-op, because of the check in the set method. 

That is correct. Here's a summary of the behavior as of right now on `main`:
- If `GlobalOpenTelemetry.get()` is called before `GlobalOpenTelemetry.set()`, an `OpenTelemetry` instance is resolved and `GlobalOpenTelemetry.set()`.
- If autoconfigure is on the classpath and autoconfigure initializes without error, `GlobalOpenTelemetry` will be set to an autoconfigured instance of `OpenTelemetrySdk`.
- If autoconfigure is not on the classpath, or some error occurs during initialization, `GlobalOpenTelemetry` will be set to `OpenTelemetry.noop()`. 

> I think this is wrong because we do want people to be able to set after calling a uninitialized get.

I'm not so sure. The current behavior is inconvenient because it forces users to either call `GlobalOpenTelemetry.set` OR have autoconfiguration do so before any call to `GlobalOpenTelemetry.get`. Trying to call `GlobalOpenTelemetry.set` later triggers an exception which can only be resolved by adjusting app initialization ordering. But the benefit of this is that `GlobalOpenTelemetry.get` always returns the same value, regardless of the order in which its called. I.e. you can't have one caller receive a `OpenTelemetry.noop()` and another caller receive an autoconfigured `OpenTelemetrySdk`. This is a nice invariant to be able to rely on.

Reply:
Hm. I can see that side of things as well (wanting in invariant). I guess it boils down to: in the case of incorrect initialization ordering, do you want a) nothing to be emitted from the SDK at all or b) at least some things emitted, from the instruments/tracers/etc that are initialized after the SDK initialization has taken place. 

I don't know that there's an easy answer to this one. 🤔

Reply:
Yup it's a good question and is kind of a design philosophy question between fail fast or allow partial success. 

I'm prefer to the fail fast because I think the absence of all telemetry provides a really strong signal to go and fix the issue.

If we do want to change this behavior should probably do it in a different PR to separate concerns.

Reply:
I'm not sure I'll be able to make sure no one will call get before I create the OpenTelemetry object.
I will need to doublecheck this PR.

Reply:
> I'm not sure I'll be able to make sure no one will call get before I create the OpenTelemetry object.

You don't need to. If you rely on `GlobalOpenTelemetry.get()` calling `AutoConfiguredOpenTelemetrySdk.initialize()`, you will just need to opt in to that behavior with an environment variable or system property. 

The current behavior should still be possible, but it shouldn't come at the expense of other scenarios that don't want the side affect. It makes sense to the `AutoConfiguredOpenTelemetrySdk.initialize()` side affect opt-in rather than opt-out because autoconfigure users should be comfortable adding environment variables / system properties so it shouldn't be a problem be a big lift to add one more.